examples/seismic/tutorials/06_skimage_tv.ipynb

```diff
@@ -0,0 +1,416 @@
+    "# so we can easily see what is going on.\n",
+    "def get_true_model():\n",
+    "    shape, spacing, origin = get_grid()\n",
+    "    return demo_model('circle-isotropic', vp=3.0, vp_background=2.5, \n",
+    "                      origin=origin, shape=shape, spacing=spacing, nbpml=40)\n",
+    "\n",
+    "# The initial guess for the subsurface model.\n",
+    "def get_initial_model():\n",
+    "    shape, spacing, origin = get_grid()\n",
+    "\n",
+    "    return demo_model('circle-isotropic', vp=2.5, vp_background=2.5, \n",
+    "                      origin=origin, shape=shape, spacing=spacing, nbpml=40)\n",
+    "\n",
+    "\n",
+    "from examples.seismic.acoustic import AcousticWaveSolver\n",
+    "from examples.seismic import RickerSource, Receiver\n",
+    "\n",
+    "# This is used by the worker to get the current model. \n",
+    "def get_current_model(param):\n",
+    "    \"\"\" Returns the current model.\n",
+    "    \"\"\"\n",
+    "    model = get_initial_model()\n",
+    "    model.m.data[:] = np.reshape(np.load(param['model']), model.m.data.shape)\n",
+    "    return model\n",
+    "\n",
+    "# Inversion crime alert! Here the worker is creating the 'observed' data\n",
+    "# using the real model. For a real case the worker would be reading\n",
+    "# seismic data from disk.\n",
+    "def get_data(param):\n",
+    "    \"\"\" Returns source and receiver data for a single shot labeled 'shot_id'.\n",
+    "    \"\"\"\n",
+    "    true_model = get_true_model()\n",
+    "    dt = true_model.critical_dt  # Time step from model grid spacing\n",
+    "\n",
+    "    # Set up source data and geometry.\n",
+    "    nt = int(1 + (param['tn']-param['t0']) / dt)  # Discrete time axis length\n",
+    "\n",
+    "    src = RickerSource(name='src', grid=true_model.grid, f0=param['f0'],\n",
+    "                       time=np.linspace(param['t0'], param['tn'], nt))\n",
+    "    src.coordinates.data[0, :] = [30, param['shot_id']*1000./(param['nshots']-1)]\n",
+    "\n",
+    "    # Set up receiver data and geometry.\n",
+    "    nreceivers = 101  # Number of receiver locations per shot\n",
+    "    rec = Receiver(name='rec', grid=true_model.grid, npoint=nreceivers, ntime=nt)\n",
+    "    rec.coordinates.data[:, 1] = np.linspace(0, true_model.domain_size[0], num=nreceivers)\n",
+    "    rec.coordinates.data[:, 0] = 980. # 20m from the right end\n",
+    "\n",
+    "    # Set up solver - using model_in so that we have the same dt,\n",
+    "    # otherwise we should use pandas to resample the time series data. \n",
+    "    solver = AcousticWaveSolver(true_model, src, rec, space_order=4)\n",
+    "\n",
+    "    # Generate synthetic receiver data from true model\n",
+    "    true_d, _, _ = solver.forward(src=src, m=true_model.m)\n",
+    "\n",
+    "    return src, true_d, nt, solver\n",
+    "\n",
+    "# Define a type to store the functional and gradient.\n",
```

Comment:
You have been using functional, objective function and f interchangeably without making it clear anywhere that they are the same thing. I think it would be a good idea to to use one of either functional or objective function everywhere, and define `f` to be the symbol for that somewhere explicitly.

Reply:
Now defined in the introduction of tutorial 3.